prow/plank/controller.go

```diff
@@ -135,6 +135,25 @@ func (c *Controller) canExecuteConcurrently(pj *prowapi.ProwJob) bool {
 		c.log.WithFields(pjutil.ProwJobFields(pj)).Debugf("Not starting another instance of %s, already %d running.", pj.Spec.Job, numPending)
 		return false
 	}
+
+	var olderMatchingPJs []prowapi.ProwJob
+	for _, foundPJ := range c.pjs {
```

Comment:
Please use the `c.pjLock` (and define a lock ordering with `c.lock` to avoid deadlock) or leave a comment explaining why you aren't using it.